### PR TITLE
[Release] Cherrypick #4350 and #4462 into Release branch

### LINKED ISSUE #4471: Release 7.0 branch problems
As you may have noticed the release branch had some problems due to an incorrect merge that I forced without noticing.
I've rebuilt the branch an applied all hotfixes already submitted without the problematic one.

As I side effect all the open PR targeting the old Release-7.0 tag were automatically closed, so I recreated them as well. Please see #4468, #4469 and #4470 (#4451 was left behind as should be no longer necessary)

Also @loumalouomega @RiccardoRossi @philbucher @KlausBSautter you will need to remake the hotfix in #4350 as it is the one that was causing the problems.

I apologize for the problems.

### PR DESCRIPTION
**Pull request by loumalouomega**
Alsso cherrypicks #4197 
Fixes #4471

### CODE REVIEW

**Comment by roigcarlo** [comment]
We are stuck with the same problem. Tests fail after adding this.

**Comment by roigcarlo** [comment]
@loumalouomega Idea. Can we split this into multiple smaller PR to see exactly at which point things start to fail?

**Comment by loumalouomega** [comment]
> @loumalouomega Idea. Can we split this into multiple smaller PR to see exactly at which point things start to fail?

I don't think that is the problem, the PR are "self contained"

**Comment by loumalouomega** [comment]
Certantly the cpp tests of the adjoints takes long time @msandre can you take a look?

**Comment by roigcarlo** [comment]
But it makes no sense:
- This works in the master and this are the same commits that the master has.
- The release branch contains nothing but cherry-picks from the master. 

If the behavior is different, then something in the master that is not here is required for the tests to work, even of it is not part of the original PR.

**Comment by loumalouomega** [comment]
> But it makes no sense:
> 
>     * This works in the master and this are the same commits that the master has.
> 
>     * The release branch contains nothing but cherry-picks from the master.
> 
> 
> If the behavior is different, then something in the master that is not here is required for the tests to work, even of it is not part of the original PR.

That's why I asked to @msandre I am not familiar with changes on Adjoint

**Comment by roigcarlo** [comment]
Dear all, any news on this?

File: applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py [hunk 1]
```diff
@@ -282,7 +253,7 @@ def _check_results_cable(self,mp,Force_X):
 
 
         self.assertAlmostEqual(disp_u_2, 0.022296019142446475,6)
-        self.assertAlmostEqual(r_u_1, -Force_X,6)
+        self.assertAlmostEqual(r_u_1, -100000000.00093779,6)
```
Comment (RiccardoRossi) [inline applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py]: shouldn't this be exactly 100000000

Comment (philbucher) [reply applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py]: yes, but you yourself asked for this change in #4286

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py]: It is  a very small change...

Comment (philbucher) [reply applications/StructuralMechanicsApplication/tests/test_patch_test_truss.py]: fixed properly

**Comment by philbucher** [comment]
I hope it works now
I commented the problematic test. It works on master, no idea why it fails here

**Comment by philbucher** [comment]
now it works

**Review by philbucher — APPROVED** [review]

**Review by RiccardoRossi — APPROVED** [review]
approving.

the "proper fix" should also go in master no?

**Comment by loumalouomega** [comment]
The master already works...

**Comment by philbucher** [comment]
> approving.
> 
> the "proper fix" should also go in master no?

I just merged it into master

**Pull request merged**